druid/src/widget/slider.rs

```diff
@@ -68,7 +88,14 @@ impl Slider {
         let scalar = ((mouse_x + self.x_offset - knob_width / 2.) / (slider_width - knob_width))
             .max(0.0)
             .min(1.0);
-        self.min + scalar * (self.max - self.min)
+        let mut value = self.min + scalar * (self.max - self.min);
+        if let Some(stepping) = self.step {
+            // Determine the number of steps and fit the slider position to a discrete step
+            let steps = (self.max / stepping).round();
+            let step = (value * steps / self.max).round();
+            value = step / steps * self.max;
```

Comment:
I think, this is wrong

say `self.min = 5`, `self.max = 25`, `stepping = 10` and `value = 13`

so 
`steps = 25/10 = (2.5).round() = 3`
`step = (13 * 3 / 25).round()` = `1.56` = `2`
`value = 2 / 3 * 25 = 16.66...`

but the value should be `15`

(sorry if maths is wrong :grimacing: )

Reply:
Why should the value be `15`?
`25` % `10` != `0`, so it seems logical to me that the stepper values can be fractional. Let's list the possible steps:

`step = 0` => `value = 0 / 3 * 25 = 0.0`
`step = 1` => `value = 1 / 3 * 25 = 8.3..`
`step = 2` => `value = 2 / 3 * 25 = 16.66`
`step = 3` => `value = 3 / 3 * 25 = 25`

As you can see, this yields a uniform step length of `8.33..`, which is what I would expect.

Reply:
the value should start from `5`(self.min) and step by `10` and end at `25`
so, 5, 15, 25 should be the possible values :shrug: